Refactor `NewWindow` by using struct

tsu-ku-ne opened the pull request:
* Passing parameters to window constructor by using struct.
* Support `has_frame`.  
https://github.com/breach/thrust/blob/master/docs/api/window.md

miketheprogrammer commented:
I love it, however I have to leave it as a pull request until I have time to change the examples. 
Unless you would like to refactor the examples as well.

I will hopefully be resuming work on this for now. Until then, or until you refactor the examples, I hope people whom need this functionality will see this pull request and pull your fork.

Thanks for the work.

tsu-ku-ne commented:
@miketheprogrammer Thanks for checking.

I changed examples and tutorials.
Adding to the above, I changed `has_frame` to conform specification below. 

> https://github.com/breach/thrust/blob/master/docs/api/window.md
> * `has_frame` creates a frameless window if `true`

miketheprogrammer commented:
Merging. Thanks.
